### ISSUE TITLE
Padding Token in Transformer XL

### ISSUE THREAD

**Issue by sb1992** [huggingface/pytorch-pretrained-BERT#586]
I have sentences of varying lengths and I was wondering how to handle that as I could not see any padding token present. The index 0 refers to <eos> in the vocab, so any help on addition of padding would be appreciated

**Comment by thomwolf** [comment]
For these causal models that consider the left-context only, it's ok not to worry too much about padding since the attention modules only look to the previous tokens. Just be careful when you compute the loss to ignore the out-of-sentence-tokens (using loss functions `ignore_index` for instance).

**Comment by bajajahsaas** [comment]
Hi, I have a related doubt. In the example code [run_language_modeling.py](https://github.com/huggingface/transformers/blob/master/examples/run_language_modeling.py), GPT-2 and transformer-xl throw error due to lacking padding token.

Please check recent comments in this [Issue](https://github.com/huggingface/transformers/issues/3021)
